src/core/lib/draw-layers.js

```diff
@@ -239,11 +239,11 @@ function updateLayerHighlightColor(layer) {
   // TODO - inefficient to update settings every render?
   // TODO: Add warning if 'highlightedObjectIndex' is > numberOfInstances of the model.
 
-  if (layer.state.model) {
-    const pickingSelectedColorValid = layer.props.highlightedObjectIndex >= 0;
-    const pickingSelectedColor = pickingSelectedColorValid ?
-      layer.encodePickingColor(layer.props.highlightedObjectIndex) :
-      layer.nullPickingColor();
+  // Update picking module settings if highlightedObjectIndex is set.
+  // This will overwrite any settings from auto highlighting.
+  const pickingSelectedColorValid = layer.props.highlightedObjectIndex >= 0;
+  if (layer.state.model && pickingSelectedColorValid) {
```

Comment:
Concern: What happens when we transition from valid to invalid? Won't the module settings need to be updated? This change seems to restrict updates to the valid case?

Reply:
During picking cycle, we update `pickingSelectedColorValid` to true/false based on valid/invalid transition. Here we overwrite that if `highlightedObjectIndex` is set.